[IMP] fleet: new vehicle updates (#8162)
odoo/documentation: content/applications/hr/fleet/new_vehicle.rst

============
New vehicles
============

Odoo's *Fleet* app manages all vehicles and the accompanying documentation that comes with vehicle
maintenance and driver's records.

All vehicles are organized within the main :guilabel:`Fleet` dashboard. Every vehicle has its own
*vehicle form*, which is displayed as a card in the Kanban view, according to its status. Each
vehicle form is displayed in the corresponding Kanban stage based on its status. The default stages

> are :guilabel:`New Request`, :guilabel:`To Order`, :guilabel:`Registered`, and

Author: I checked with the experts and then tested in a blank DB with only fleet installed- those columns appear because they were added/configured in the runbot and are not actual default columns that appear. I also thought they were default columns, and were in my previous documentation, but I was wrong.

To add a new vehicle to the fleet, click the :guilabel:`New` button in the top-left corner, and a
blank vehicle form loads. Enter the vehicle information in the vehicle form.

The form auto-saves as data is entered. However, the form can be saved manually at any time by
clicking the :guilabel:`Save manually` option, represented by a :guilabel:`cloud with an upwards

> arrow inside` icon.

Author: Thank you! This was a change from KSC, and I have used this in a lot of my docs ever since- but I like this definition a lot better.

============
New vehicles
============

Odoo's *Fleet* app manages all vehicles, and the accompanying documentation that comes with vehicle
maintenance, and drivers records.

> All vehicles are organized within the :guilabel:`Vehicles` dashboard, which is the default dashboard
> for the *Fleet* application. Each vehicle is displayed in its corresponding Kanban stage, based on
> its status. The default stages are :guilabel:`New Request`, :guilabel:`To Order`,
> :guilabel:`Registered`, and :guilabel:`Downgraded`.

Reviewer: Small fix to navigate to the Fleet app before using a guilabel :)
```suggestion
Upon opening the :menuselection:`Fleet` application, all vehicles are organized within the
:guilabel:`Vehicles` dashboard, which is the default dashboard for the *Fleet* application. Each
vehicle is displayed in its corresponding Kanban stage, based on its status. The default stages are
:guilabel:`New Request`, :guilabel:`To Order`, :guilabel:`Registered`, and :guilabel:`Downgraded`.
```

To add a new vehicle to the fleet from the :guilabel:`Vehicles` page, click the :guilabel:`New`
button in the top-left corner, and a blank vehicle form loads. Then, proceed to enter the vehicle
information on the vehicle form.

The form auto-saves as data is entered. However, the form can be saved manually at any time by

> clicking the :guilabel:`Save manually` option, represented by a :guilabel:`cloud upload` icon,
> located in the top-left corner of the page.

Reviewer: Icon formatting
```suggestion
clicking the *save manually* option, represented by a :guilabel:`(cloud upload)` icon, located in
the top-left corner of the page.
```

any plans for a change in the driver in the future, and when.

- :guilabel:`Driver`: select the driver from the drop-down menu, or type in a new driver and click
  either :guilabel:`Create "driver"` or :guilabel:`Create and edit...` to :ref:`create a new driver
  and edit the driver details <fleet/new_vehicle/add-driver>`.

  .. important::
     A driver does **not** have to be an employee. All drivers are stored within the *Fleet*

>      application, and if the *Contacts* application is installed, they are also stored there.

Author: Same- good point! I had to test this a few times to make sure I understood this as well, since it's really odd that I couldn't access specific lists within an app.

     When creating a new driver, the driver is added to both the *Fleet* application and the
     *Contacts* application, **not** the *Employees* application.

     If the *Contacts* application is not installed when a driver is added, the data is stored
     within the *Fleet* application. Upon installation of the *Contacts* application, all drivers

>      created in the *Fleet* application are also visible in the *Contacts* application.

Reviewer 2: There seems to be a bit of duplicated information here -- would you mind cleaning this up a bit, in order to more succinctly summarize this portion? Thanks!

- :guilabel:`Mobility Card`: if the selected driver has a mobility card listed on their employee
  card in the *Employees* application, the mobility card number appears in this field. If there is

>   no mobility card listed and one should be added, :ref:`edit the employee record

Reviewer 2: ```suggestion
  no mobility card listed, and one should be added, :ref:`edit the employee record
```

  <employees/hr-settings>` in the *Employees* application.
- :guilabel:`Future Driver`: if the next driver for the vehicle is known, select the next driver
  from the drop-down menu, or type in the next driver and click either :guilabel:`Create "future
  driver"` or :guilabel:`Create and edit...` to :ref:`create a new future driver and edit the driver
  details <fleet/new_vehicle/add-driver>`.
- :guilabel:`Plan To Change Car`: if the current driver set for this vehicle plans to change their
  vehicle, either because they are waiting on a new vehicle that is being ordered, or this is a
  temporary vehicle assignment, and they know which vehicle they are driving next, check this box.
  Do **not** check this box if the current driver does not plan to change their vehicle.
- :guilabel:`Assignment Date`: using the drop-down calendar, select when the vehicle is available
  for another driver. Select the date by navigating to the correct month and year using the
  :guilabel:`⬅️ (left arrow)` and :guilabel:`➡️ (right arrow)` icons, then click on the specific

>   day. If this field is blank, this indicates the vehicle is currently available and can be

Reviewer 2: ```suggestion
  day. If this field is blank, this indicates the vehicle is currently available, and can be
```

  :guilabel:`Follow-up Address`, or :guilabel:`Other Address`.

  If desired, select one of these other options for the :guilabel:`Contact` field, and enter the
  corresponding information.

  If :guilabel:`Company` is selected for the  :guilabel:`Individual` or :guilabel:`Company` field,
  this field is labeled :guilabel:`Address` and cannot be modified.

- :guilabel:`Tax ID`: enter the driver or company's tax ID in this field.
- :guilabel:`Job Position`: enter the driver's job position in this field. If :guilabel:`Company` is
  selected for the :guilabel:`Individual` or :guilabel:`Company` field, this field does not appear.
- :guilabel:`Phone`: enter the driver or company's phone number in this field.
- :guilabel:`Mobile`: enter the driver or company's mobile number in this field.
- :guilabel:`Email`: enter the driver or company's email address in this field.
- :guilabel:`Website`: enter the driver or company's website address in this field.
- :guilabel:`Title`: using the drop-down menu, select the driver's title in this field. The default
  options are :guilabel:`Doctor`, :guilabel:`Madam`, :guilabel:`Miss`, :guilabel:`Mister`, and
  :guilabel:`Professor`.

> 

Reviewer 2: I'm also seeing a Language field too -- does that not appear for you?

Author: That's from the Employees app (I believe).

  If :guilabel:`Company` is selected for the :guilabel:`Individual` or :guilabel:`Company` field,
  this field does not appear.
- :guilabel:`Tags`: using the drop-down menu, select any tags that apply to the driver or company.

  To add a new tag, type in the tag, then click :guilabel:`Create "tag"`.

  There is no limit to the number of tags that can be selected.

.. image:: new_vehicle/create-driver.png
   :align: center
   :alt: The top portion of the create driver form.

Contacts & addresses tab
************************

After completing the top-half of the :guilabel:`Create Driver` or :guilabel:`Create Future Driver`
form, add any other contacts and addresses associated with the driver or company in this tab.

To add a new contact, click the :guilabel:`Add` button, and a :guilabel:`Create Contact` pop-up
window appears.

Enter the following information on the form:

> - :guilabel:`Contact Type`: select the type of contact being added. Click the radio button next to

Reviewer 2: i understand your reasoning here, but i'm not sure if this should have GUI, as it doesn't directly appear this way on the UI. Perhaps there's a different way to mention it, so as to keep in line with the established RST guidelines.

Author: This was a hard one! I know in other docs, it was suggested to have a gui in bullet lists even for an entry like this, so that it looks nice when reading it- even though it is a technical stretch for a gui label.  I removed it- it doesn't look too bad.

  the desired type. The default options are:

  - :guilabel:`Contact`: select this option to add general contact details for employees of the
    associated company.
  - :guilabel:`Invoice Address`: select this option to add a preferred address for all invoices.
    When added to the form, this address is selected by default when sending an invoice to the
    associated company.
  - :guilabel:`Delivery Address`: select this option to add a preferred address for all deliveries.
    When added to the form, this address is selected by default when delivering an order to the
    associated company.
  - :guilabel:`Follow-up Address`: select this option to add a preferred address for all follow-up
    correspondence. When added to the form, this address is selected by default when sending
    reminders about overdue invoices.
  - :guilabel:`Other Address`: select this option to add any other necessary addresses for the

>     company.

Reviewer 2: ```suggestion
    company or driver.
```
right?

Author: Yes! You are correct

.. image:: new_vehicle/create-contact.png
   :align: center
   :alt: The create contact form with all parts filled in.

Depending on the :guilabel:`Contact Type`, some optional fields may not be visible. The available
fields are identical to the fields in the :ref:`general information
<fleet/new_vehicle/general-info>` section of the new driver form.

Add any notes to the :guilabel:`Internal notes...` section of the form.

After entering all of the information, click either :guilabel:`Save & Close` to add the one new
contact, or :guilabel:`Save & New` to add the current address record and create another address
record.

As contacts are added to this tab, each contact appears in a separate box, with an icon indicating
what type of contact is listed.

.. example::

>    An :guilabel:`Invoice Address` displays a dollar sign icon inside that specific address box,

Reviewer 2: is there any dollar sign emoji that could possibly be used here?

Author: Sure! It didn't occur to me since I thought of that as a universally understood icon, but you are right- a nice add.

>    whereas a :guilabel:`Delivery Address` displays a truck icon inside.

Reviewer 2: same emoji-related suggestion here

   .. image:: new_vehicle/contacts-address.png
     :align: center
     :alt: The create contact form with all parts filled in.

Sales & purchase tab
********************

> Enter the following sales and purchasing information for the various sections below. Depending on

Reviewer 2: ```suggestion
Enter the following sales and purchasing information, in the :guilabel:`Sales & Purchase` tab of the :guilabel:`Create Driver` or :guilabel:`Create Future Driver` pop-up form, for the various sections below. Depending on
```

the other installed applications, additional fields and sections may appear. The following are all

> default fields for the *Fleet* application only.

Reviewer 2: ```suggestion
default fields for the *Fleet* application **only**.
```

> Sales

Reviewer 2: ```suggestion
Sales section
```

^^^^^

> - :guilabel:`Salesperson`: using the drop-down menu, select the person who is the main point of

Reviewer 3: ```suggestion
- :guilabel:`Salesperson`: using the drop-down menu, select the user who is the main point of
```

  contact for sales with this driver's company.

  This person **must** be an internal user of the company, meaning they can log into the database as
  a user.

> 

Reviewer 2: I'm also seeing fields for: Payment Terms, Pricelist, Avalara Code, Delivery Method...

...so please double-check to ensure all the necessary fields are being mentioned/explained. Thanks!

Author: I had a soul-searching moment with this, because you are correct- in various Runbots, with ALL the apps installed, there are a lot of other fields and tabs that are present. So after first making a VERY long (too long in my opinion) version of this section, I then made a DB to see what was actually in a Fleet install without other items. So I went with that route- and why I had added that 'this is only the defaults for fleet' statement.

> Misc

Reviewer 2: ```suggestion
Misc section
```
There was a 'Point of Sale' section between 'Sales' and 'Misc' -- worth a double-check into this, as well. Thanks!

Author: That's from having the sales app installed- and there are even MORE tabs if you have all apps and lots of added configurations. That's why I went with the more brief version for only the fleet app, and made that statement before getting into the weeds about these being only the default items for fleet.

^^^^

> - :guilabel:`Company ID`: if the company has an ID number **other than** its *tax ID*, enter it in

Reviewer 2: ```suggestion
- :guilabel:`Company ID`: if the company has an ID number, **other than** its *tax ID*, enter it in
```

  this field.

> - :guilabel:`Reference`: enter any text to give more information regarding the contact person. This

Reviewer 2: There was a 'Citizen Identification' field beneath the Company ID field. Might be worth double-checking into this, too

Author: That's being pulled from the Employees app. There are many fields that appear as a result of other apps installed.

  is an internal note to provide any additional information.

  .. example::
     A company has several people with the same name, John Smith. The :guilabel:`Reference` field
     could state `John Smith at X205 - purchaser` to provide additional details.

> 

Reviewer 2: Beneath the 'Reference' field, there was a Company field, Website field, and Website Tags field.

Reviewer 2: Also there was an entire Purchase section of fields and a brief Fiscal Information field that appeared on this form, too

Author: That's from the purchase app and the localization settings made in there (which I chose to leave out to prevent this form being 2k lines, etc, and kept this to the default Fleet app parts that appear with Fleet and nothing else).

> Internal notes tab

Reviewer 2: There was an Accounting tab between Sales & Purchase and Internal Notes -- at least on the couple runbots I checked during my review. Might be worth looking into...

Author: That's from the Accounting app being installed.

******************

> Add any notes that pertain to the driver or any other necessary information, in this tab.

Reviewer 2: There was also a field labeled: 'Warning on the Sales Order' available in this tab of the 'create driver' pop-up form

And there was a 'Warning on the Invoice' field, too

Do those typically *not* appear? If they do, it might be worth mentioning their purpose

Author: Again - there are many other fields that are linked to specific apps only (these appear from the sales app). I had mentioned in the beginning of the form sections that this was only the items in the default fleet app- but I can add more mentions if you think it is necessary? I didn't want to add an identical note to each section, which is why I kept  it to the first mention only.

> Vehicle section

Reviewer 2: You appear to jump away from the 'Create Driver' form breakdown and *back* to the 'new form' breakdown here -- however, in the runbots I'm using for my review, there are 2 more tabs in t he 'Create Driver' pop-up that haven't been focused on up to this point: Partner Assignment and Membership

Can you please double-check and add explanations about these tabs, as well, if needed? Thanks!

Author: Those are also from other apps (I do not know off the top of my head which ones make that appear, but they do not appear for a fleet-only DB).

  options are :guilabel:`Doctor`, :guilabel:`Madam`, :guilabel:`Miss`, :guilabel:`Mister`, and
  :guilabel:`Professor`.

  If the :guilabel:`Company` radio button is selected at the top of the form, this field does not
  appear.
- :guilabel:`Tags`: using the drop-down menu, select any tags that apply to the driver or company.

  To add a new tag, type in the tag, then click :guilabel:`Create "tag"`.

  There is no limit to the number of tags that can be selected.

.. image:: new_vehicle/create-driver.png
   :align: center
   :alt: The top portion of the create driver form.

Contacts & Addresses tab
************************

After completing the top-half of the :guilabel:`Create Driver` or :guilabel:`Create Future Driver`
form, add any other contacts and addresses associated with the driver or company in this tab.

To add a new contact, click the :guilabel:`Add` button, and a :guilabel:`Create Contact` pop-up
window appears.

Before entering the necessary information on the form, select the type of contact being added from a
series radio button options located at the top of the form. Those options are:

>   - :guilabel:`Contact`: select this option to add general contact details for employees of the
>     associated company.
>   - :guilabel:`Invoice Address`: select this option to add a preferred address for all invoices.
>     When added to the form, this address is selected by default when sending an invoice to the
>     associated company.
>   - :guilabel:`Delivery Address`: select this option to add a preferred address for all deliveries.
>     When added to the form, this address is selected by default when delivering an order to the
>     associated company.
>   - :guilabel:`Follow-up Address`: select this option to add a preferred address for all follow-up
>     correspondence. When added to the form, this address is selected by default when sending
>     reminders about overdue invoices.
>   - :guilabel:`Other Address`: select this option to add any other necessary addresses for the
>     company or driver.

Reviewer: No need for these to be indented :)
```suggestion
- :guilabel:`Contact`: select this option to add general contact details for employees of the
  associated company.
- :guilabel:`Invoice Address`: select this option to add a preferred address for all invoices.
  When added to the form, this address is selected by default when sending an invoice to the
  associated company.
- :guilabel:`Delivery Address`: select this option to add a preferred address for all deliveries.
  When added to the form, this address is selected by default when delivering an order to the
  associated company.
- :guilabel:`Follow-up Address`: select this option to add a preferred address for all follow-up
  correspondence. When added to the form, this address is selected by default when sending
  reminders about overdue invoices.
- :guilabel:`Other Address`: select this option to add any other necessary addresses for the
  company or driver.
```

  selected for the :guilabel:`Individual` or :guilabel:`Company` field, this field does not appear.
- :guilabel:`Phone`: enter the driver or company's phone number in this field.
- :guilabel:`Mobile`: enter the driver or company's mobile number in this field.
- :guilabel:`Email`: enter the driver or company's email address in this field.
- :guilabel:`Website`: enter the driver or company's website address in this field.
- :guilabel:`Title`: using the drop-down menu, select the driver's title in this field. The default
  options are :guilabel:`Doctor`, :guilabel:`Madam`, :guilabel:`Miss`, :guilabel:`Mister`, and
  :guilabel:`Professor`.

  If :guilabel:`Company` is selected for the :guilabel:`Individual` or :guilabel:`Company` field,
  this field does not appear.
- :guilabel:`Tags`: using the drop-down menu, select any tags that apply to the driver or company.

  To add a new tag, type in the tag, then click :guilabel:`Create "tag"`.

  There is no limit to the number of tags that can be selected.

.. image:: new_vehicle/create-driver.png
   :align: center
   :alt: The top portion of the create driver form.

> Contacts & addresses tab

Author: I agree- and I changed it in yesterdays doc you reviewed, so this was between that edit and this one =)

************************

After completing the top-half of the :guilabel:`Create Driver` or :guilabel:`Create Future Driver`
form, add any other contacts and addresses associated with the driver or company in this tab.

To add a new contact, click the :guilabel:`Add` button, and a :guilabel:`Create Contact` pop-up
window appears.

> Enter the following information on the form:
> 
> - Contact Type: select the type of contact being added. Click the radio button next to the desired
>   type. The default options are:

Reviewer 2: ```suggestion
Before entering the necessary information on the form, select the type of contact being added from a
series radio button options located at the top of the form. Those options are:
```

    company or driver.

.. image:: new_vehicle/create-contact.png
   :align: center
   :alt: The create contact form with all parts filled in.

Depending on the :guilabel:`Contact Type`, some optional fields may not be visible. The available
fields are identical to the fields in the :ref:`general information
<fleet/new_vehicle/general-info>` section of the new driver form.

Add any notes to the :guilabel:`Internal notes...` section of the form.

After entering all of the information, click either :guilabel:`Save & Close` to add the one new
contact, or :guilabel:`Save & New` to add the current address record and create another address
record.

As contacts are added to this tab, each contact appears in a separate box, with an icon indicating
what type of contact is listed.

.. example::
   An :guilabel:`Invoice Address` displays a :guilabel:`💲 (dollar sign)` icon inside that specific
   address box, whereas a :guilabel:`Delivery Address` displays a :guilabel:`🚚 (truck)` icon
   inside.

   .. image:: new_vehicle/contacts-address.png
     :align: center
     :alt: The create contact form with all parts filled in.

> Sales & purchase tab

Reviewer 2: same note as above

********************

Enter the following sales and purchasing information, in the :guilabel:`Sales & Purchase` tab of the

> :guilabel:`Create Driver` or :guilabel:`Create Future Driver` pop-up form, for the various sections

Reviewer 2: ```suggestion
:guilabel:`Create Driver` or :guilabel:`Create Future Driver` pop-up form for the various sections
```

below.

Depending on the other installed applications, additional fields and sections may appear. The
following are all default fields for the *Fleet* application **only**.

Sales section
^^^^^^^^^^^^^

- :guilabel:`Salesperson`: using the drop-down menu, select the user who is the main point of
  contact for sales with this driver's company.

  This person **must** be an internal user of the company, meaning they can log into the database as
  a user.

Misc
^^^^

- :guilabel:`Company ID`: if the company has an ID number, **other than** its *tax ID*, enter it in
  this field.
- :guilabel:`Reference`: enter any text to give more information regarding the contact person. This
  is an internal note to provide any additional information.

  .. example::
     A company has several people with the same name, John Smith. The :guilabel:`Reference` field
     could state `John Smith at X205 - purchaser` to provide additional details.

> Internal notes tab

Reviewer 2: same note as above

******************

> Add any notes that pertain to the driver or any other necessary information, in this tab.

Reviewer 2: ```suggestion
Add any notes that pertain to the driver, or any other necessary information, in this tab.
```

Vehicle section
---------------

This section of the vehicle form relates to the physical details of the vehicle.

If a preexisting vehicle in the database was selected for the :guilabel:`Model` field in the top
portion of the form, some fields may auto-populate, and additional fields may also appear.

Fill in the following fields on the form:

- :guilabel:`Category`: using the drop-down menu, select the vehicle category from the available
  options. To create a new category, type in the new category name, then click :guilabel:`Create
  "category"`.
- :guilabel:`Order Date`: using the drop-down calendar, select the date the vehicle was ordered.

> - :guilabel:`Registration Date`:  using the drop-down calendar, select the date the vehicle was

Reviewer 2: ```suggestion
- :guilabel:`Registration Date`: using the drop-down calendar, select the date the vehicle was
```

  registered.
- :guilabel:`Cancellation Date`: using the drop-down calendar, select the date the vehicle lease
  expires, or when the vehicle is no longer available.
- :guilabel:`Chassis Number`: enter the chassis number in the field. This is known in some countries
  as the :abbr:`VIN (Vehicle Identification Number)` number.
- :guilabel:`Last Odometer`: enter the last known odometer reading in the number field. Using the
  drop-down menu next to the number field, select whether the odometer reading is in kilometers
  :guilabel:`(km)` or miles :guilabel:`(mi)`.
- :guilabel:`Fleet Manager`: select the fleet manager from the drop-down menu, or type in a new

>   fleet manager and click either :guilabel:`Create` or :guilabel:`Create and Edit`.

Reviewer 2: ```suggestion
  fleet manager, and click either :guilabel:`Create` or :guilabel:`Create and Edit`.
```

- :guilabel:`Location`: type in the specific location where the vehicle is typically located in this
  field. The entry should clearly explain where the vehicle can be found, such as `Main Garage` or
  `Building 2 Parking Lot`.

.. image:: new_vehicle/new-vehicle-type.png
   :align: center
   :alt: The new vehicle form, showing the vehicle tax section.

> Tax info tab

Reviewer 2: same note as above

------------

Depending on the localization setting for the database, and what additional applications are
installed, other fields may be present on the form.

> The sections below are default and appear for all vehicles, regardless of what other applications
> are installed, or the localization setting.

Reviewer 2: ```suggestion
The sections below are default and appear for all vehicles, regardless other installed applications or localization settings.
```

- :guilabel:`Residual Value`: enter the current value of the vehicle.

.. note::

>    The values listed above affects the accounting department. It is recommended to check with the

Reviewer 3: ```suggestion
   The values listed above affect the accounting department. It is recommended to check with the
```

---------

If the model for the new vehicle is already configured in the database, the :guilabel:`Model` tab

> populates the corresponding information. If the model is **not** already in the database and the

Reviewer 2: ```suggestion
populates the corresponding information. If the model is **not** already in the database, and the
```

> :guilabel:`Model` tab needs to be configured, it is recommended to :ref:`configure the new vehicle
> model <fleet/configuration/add-model>` so this information auto-populates.

Reviewer 3: ```suggestion
:guilabel:`Model` tab needs to be configured, it is recommended that :ref:`the new vehicle
model <fleet/configuration/add-model>` be configured so this information auto-populates.
```
